File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = dict()
        for x in strs:
            word_dict = dict()
            for c in x:
                if c in word_dict:
                    count = word_dict[c]
                    word_dict.update({c:count+1})
                else:
                    word_dict[c] = 1

            key = tuple(sorted(word_dict.items()))

            if key in result:
                group = result[key]
                group.append(x)
            else:
                result[key] = [x]   

        return list(result.values())                
                      

        

File: Data Structures & Algorithms/longest-common-prefix/submission-2.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        String result = "";

        for(int i = 0; i < strs[0].length(); i++) {
            String str = strs[0].substring(0,i+1);
            for(int j = 1; j < strs.length; j++) {
                if(strs[j].length() < str.length() || !strs[j].startsWith(str)) {
                    return result;
                }
            }
            result = str;
        }

        return result;
    }
}